synchronization/semaphore.cc

```diff
@@ -48,8 +48,48 @@ class PlatformSemaphore {
 
 }  // namespace flutter
 
-#else  // OS_MACOSX
+#elif OS_WIN
+#include <windows.h>
+
+namespace flutter {
+
+class PlatformSemaphore {
+ public:
+  explicit PlatformSemaphore(uint32_t count)
+      : _sem(CreateSemaphore(NULL, count, 1024, NULL)) {}
+
+  ~PlatformSemaphore() {
+    if (_sem != nullptr) {
+      CloseHandle(_sem);
+      _sem = nullptr;
+    }
+  }
+
+  bool IsValid() const { return _sem != nullptr; }
+
+  bool TryWait() {
+    if (_sem == nullptr) {
+      return false;
+    }
+
+    return (WaitForSingleObject(_sem, 0) == 0);
```

Comment:
Given that the API docs specify a set of `#define`d return values, consider:
```
return WaitForSingleObject(_sem, 0) == WAIT_OBJECT_0;
```

Reply:
Ah, good catch! Will update.